src/controllers/dynakube/version/updater.go

```diff
@@ -151,7 +146,6 @@ func updateVersionStatusForTenantRegistry( //nolint:revive
 	imageUri string,
 	imageVersionFunc ImageVersionFunc,
 	dockerCfg *dockerconfig.DockerConfig,
-	dynakube *dynatracev1beta1.DynaKube,
```

Comment:
is `DockerConfig` still necessary?
AFAIK the `keychain` does what the `DockerConfig` is used for.
- In every change you have you just use the `DockerConfig` as a holder for the `Dynakube` and `ApiReader`, so it feels a bit redundant.

check https://github.com/Dynatrace/dynatrace-operator/blob/ad2c7d1a66a1b40faf3835303b4ac9e5d3703718/src/controllers/dynakube/version/reconciler.go#L59
I don't think that is necessary as you are doing it [here](https://github.com/Dynatrace/dynatrace-operator/blob/ad2c7d1a66a1b40faf3835303b4ac9e5d3703718/src/controllers/dynakube/version/image_hash.go#L61-L62) aswell (just better)

so for this function (and others that use the `DockerConfig` in this package) I would pass in the dynakube and apiReader explicitly as they are needed.

tldr: I would try to remove the `DockerConfig` from the `version` package in general. (other packages will do the same when they move to the new library.)

Reply:
I agree, good point! Let me try to refactor it :+1:

Reply:
Fixed in [this commit](https://github.com/Dynatrace/dynatrace-operator/pull/1998/commits/c23e53e66bb19aaa680e8fd01e05fad805afd5a1)